vms/platformvm/state/staker.go

```diff
@@ -83,19 +83,26 @@ func (s *Staker) Less(than *Staker) bool {
 	return bytes.Compare(s.TxID[:], than.TxID[:]) == -1
 }
 
-func NewCurrentStaker(txID ids.ID, staker txs.Staker, potentialReward uint64) (*Staker, error) {
+func NewCurrentStaker(
+	txID ids.ID,
+	staker txs.PostDurangoStaker,
+	startTime time.Time,
```

Comment:
This is probably the main point of this PR, along with the PostDurangStaker type introduction. Upon drop of pending stakers, we'll explicitly pass chainTime here to create the staker object. In this PR we anticipate the structural change.